rumqttd/src/router/routing.rs

```diff
@@ -1129,16 +1239,57 @@ fn forward_device_data(
         return ConsumeStatus::FilterCaughtup;
     }
 
-    // Fill and notify device data
-    let forwards = publishes.into_iter().map(|(mut publish, offset)| {
-        publish.qos = protocol::qos(qos).unwrap();
-        Forward {
-            cursor: offset,
-            size: 0,
-            publish,
-        }
+    // there are three cases
+    // 1. we are using an alias for filter already
+    // 2. we can't use alias as none is avaliable to use
+    // 3. we need to set the alias as this is first time
+    // We want to clear topic in only in case 1!
+    let mut clear_topic = true;
+    let topic_alias = broker_topic_aliases.as_mut().and_then(|aliases| {
```

Comment:
setting `clear_topic` to true by default will cause topic to be cleared even when we are not using topic alias

Reply:
hey, please check updated code! 
and regarding this in older code, we first check if `topic_alias` is `Some()` and then only proceed further, so we aren't using topic alias, it will be `None` and clear_topic won't take effect.